include/fe/inf_fe.h

```diff
@@ -528,8 +534,229 @@ class InfFE : public FEBase
   { libmesh_assert(radial_qrule); return _n_total_qp; }
 
 
+
+  const std::vector<Point> & get_xyz () const override
+  { calculate_map = true; return xyz; }
+
+  /**
+   * Please use \p get_JxWxdecay_sq() instead!
+   *
+   * The Jacobian cannot be computed for Infinite elements!
+   *
+   */
+  const std::vector<Real> & get_JxW () const override
+  {
+      libmesh_not_implemented();
+      return this->JxWxdecay;
+  }
+
+  /**
+   * \returns Jacobian times quadrature weight times square of the
+   *  decaying function \f$ decay= r^{-\frac{dim+1}{2}}\f$
+   *
+   * This function is the variant of \p get_JxW() for \p InfFE.
+   * Since J diverges there, a respectize decay-function must be
+   * applied to obtain well-defined quantities.
+   */
+  const std::vector<Real> & get_JxWxdecay_sq () const override
+  { calculate_map = true; return this->JxWxdecay;}
+
+
+  /**
+   * \returns The shape function \p phi weighted by r/decay
+   * where \f$ decay = r^{-\frac{dim+1}{2}} \f$
+   *
+   * To compensate for the decay function applied to the Jacobian (see \p get_JxWxdecay_sq),
+   * the wave function \p phi should be divided by  this function.
+   *
+   * The factor r must be compensated for by the Sobolev \p weight.
+   * (i.e. by using \p get_Sobolev_weightxR_sq())
+   **/
+  const std::vector<std::vector<OutputShape>> & get_phi_over_decayxR () const override
+  { libmesh_assert(!calculations_started || calculate_phi);
+    calculate_phi = true; return phixr; }
+
+
+  /**
+   * \returns the gradient of the shape function (see \p get_dphi()),
+   * but in case of \p InfFE, weighted with r/decay.
+   * See \p  get_phi_over_decayxR() for details.
+   */
+  const std::vector<std::vector<OutputGradient>> & get_dphi_over_decayxR () const override
+  { libmesh_assert(!calculations_started || calculate_dphi);
+    calculate_dphi = calculate_dphiref = true; return dphixr; }
+
+
+  /**
+   * \returns the gradient of the shape function (see \p get_dphi()),
+   * but in case of \p InfFE, weighted with 1/decay.
+   *
+   * In contrast to the shape function, its gradient stays finite
+   * when divided by the decay function.
+   */
+  const std::vector<std::vector<OutputGradient>> & get_dphi_over_decay () const override
+  { libmesh_assert(!calculations_started || calculate_dphi);
+    calculate_dphi = calculate_dphiref = true; return dphixr_sq; }
+
+
+  /**
+   * \returns The element tangents in xi-direction at the quadrature
+   * points.
+   */
+  const std::vector<RealGradient> & get_dxyzdxi() const override
+  { calculate_map = true; libmesh_not_implemented();}
+  /**
+   * \returns The element tangents in eta-direction at the quadrature
+   * points.
+   */
+  const std::vector<RealGradient> & get_dxyzdeta() const override
```

Comment:
Isn't this the same as the base class implementation? I'm not sure why you need all these overrides, probably I am missing something obvious though.

Reply:
well, the difference is that usually this calls ``_fe_map->get_dxyzdzeta()`` but for infinite elements, ``_fe_map`` is not in use (anymore)...